Bugsnag/Assets/Bugsnag/Editor/SymbolUpload/BugsnagAndroidSymbolUploader.cs

```diff
@@ -0,0 +1,246 @@
+        }
+
+        private string GetBugsnagCliDownloadUrl()
+        {
+
+            // Detect platform
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.Arm64
+                    ? _cliDownloadUrl + "arm64-macos-bugsnag-cli"
+                    : _cliDownloadUrl + "x86_64-macos-bugsnag-cli";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.X86
+                    ? _cliDownloadUrl + "i386-windows-bugsnag-cli.exe"
+                    : _cliDownloadUrl + "x86_64-windows-bugsnag-cli.exe";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return RuntimeInformation.OSArchitecture == Architecture.X86
+                    ? _cliDownloadUrl + "i386-linux-bugsnag-cli"
+                    : _cliDownloadUrl + "x86_64-linux-bugsnag-cli";
+            }
+
+            throw new PlatformNotSupportedException("Your platform is not supported for Bugsnag CLI.");
+        }
+
+        private bool DownloadBugsnagCli(string cliUrl, string downloadPath)
+        {
+            UnityEngine.Debug.Log("Starting Bugsnag CLI download process...");
+
+            var directory = Path.GetDirectoryName(downloadPath);
+            if (!Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+                UnityEngine.Debug.Log($"Created directory at {directory}");
+            }
+
+            if (File.Exists(downloadPath))
+            {
+                UnityEngine.Debug.Log($"Executable already exists at {downloadPath}, skipping download.");
+                return true;
+            }
+
+            UnityEngine.Debug.Log($"Downloading Bugsnag CLI from {cliUrl} to {downloadPath}...");
+
+            var downloadProcess = new Process()
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = "curl",
+                    Arguments = $"-L {cliUrl} --output {downloadPath}",
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true
+                }
+            };
+
+            downloadProcess.Start();
+            string downloadOutput = downloadProcess.StandardOutput.ReadToEnd();
+            string downloadError = downloadProcess.StandardError.ReadToEnd();
+            downloadProcess.WaitForExit();
+
+            if (downloadProcess.ExitCode != 0)
+            {
+                UnityEngine.Debug.LogError($"Failed to download the Bugsnag CLI tool. Error: {downloadError}");
+                return false;
+            }
+
+            UnityEngine.Debug.Log($"Bugsnag CLI tool downloaded successfully. Output: {downloadOutput}");
```

Comment:
Is the output going to be useful?

```suggestion
            UnityEngine.Debug.Log($"BugSnag CLI downloaded successfully. Output: {downloadOutput}");
```